### PR TITLE
Feature/non authentication #1313

### PR DESCRIPTION
**Pull request by kaperoo**
## Description
Created nullAuthProvider which allows to disable authentication related functionality when `"auth-provider": null` is specified in SG settings. 
Also, plugin routes can now be unauthenticated by specifying the optional `unauthorised: true` argument in plugin route settings. Plugin routes are authenticated by default.

## Testing instructions
Updated unit tests to check for redirections and remounting. Added a new test suite for nullAuthProvider.

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

## Agile board tracking
connect to #1313

### CODE REVIEW

File: src/mainAppBar/mainAppBar.component.tsx [hunk 1]
```diff
@@ -252,6 +253,9 @@ const mapStateToProps = (state: StateType): MainAppProps => ({
   showAdminPageButton:
     state.scigateway.authorisation.provider.isLoggedIn() &&
     state.scigateway.authorisation.provider.isAdmin(),
+  showUserComponent:
+    state.scigateway.authorisation.provider.constructor.name !==
```
Comment (louise-davies) [inline src/mainAppBar/mainAppBar.component.tsx:257]: Could we use `instanceof` instead here? `state.scigateway.authorisation.provider instanceof NullAuthProvider`? It's less fragile

File: src/routing/routing.component.test.tsx [hunk 2]
```diff
@@ -115,6 +116,32 @@ describe('Routing component', () => {
...
+      },
+    ];
+
+    const { asFragment } = render(<Routing />, { wrapper: Wrapper });
+
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it('renders an unauthorised route for a plugin', () => {
+    state.scigateway.authorisation.provider = new TestAuthProvider('logged in');
```
Comment (louise-davies) [inline src/routing/routing.component.test.tsx:135]: Surely to double check that unauthorised routes work right we shouldn't "log in" for this test?

File: src/routing/routing.component.tsx [hunk 3]
```diff
@@ -272,6 +253,9 @@ const mapStateToProps = (state: StateType): RoutingProps => ({
       state.scigateway.authorisation.provider.autoLogin &&
       localStorage.getItem('autoLogin') === 'true'
     ),
+  nullAuthProvider:
+    state.scigateway.authorisation.provider.constructor.name ===
```
Comment (louise-davies) [inline src/routing/routing.component.tsx:257]: Same comment as above aboutusing `instanceof`

**Review by louise-davies — COMMENTED** [review]
Functionality works fine - just want to check with @joshuadkitenge that he's happy with this for IMS. Also some minor code comments

**Comment by joshuadkitenge** [comment]
> Functionality works fine - just want to check with @joshuadkitenge that he's happy with this for IMS. Also some minor code comments

It works perfectly with IMS once I merged the changes into the react-18 branch. Great work @kaperoo

File: src/routing/routing.component.test.tsx (hunk 2 above)
Comment (kaperoo) [reply src/routing/routing.component.test.tsx]: Yes, my mistake 😅

**Review by louise-davies — APPROVED** [review]
LGTM and Josh confirms it works for IMS :)

**Pull request merged**